src/types.rs

```diff
@@ -840,6 +841,27 @@ impl Types {
         Types { pos, ..self }
     }
 
+    /// Returns the same type with the position cleared (set to `None`).
+    pub fn without_pos(self) -> Types {
+        self.traverse::<_, _, ()>(
+            &|t, _| {
+                Ok(Types {
+                    pos: TermPos::None,
+                    ..t
+                })
+            },
+            &mut (),
+            TraverseOrder::BottomUp,
+        )
+        .unwrap()
+    }
+
+    pub fn range(&self) -> Option<Range<usize>> {
```

Comment:
Why do we need this function? I think we can just store a `TermPos`, or a `RawSpan` directly inside `PathSpan`, instead of `start` and `end`. Then you can use `error::primary()` which takes a `RawSpan`. You shouldn't have to deal with low-level stuff like `start` and `end` anymore, and only manipulate `TermPos` or `RawSpan`.